### PR TITLE
Issue #141: Added <auto-generated> tagged comments for each code file

### LINKED ISSUE #141: Include // <auto-generated> at top of all generated files
### Motivation

We're attempting to use null reference checking and forcing null-reference checks to be compile errors, however generated files do not abide. 

That would be fine, as warnings are usually ignored in generated files but the KontentModelGenerator currently does not mark them as autogenerated, and there is no option to enable such a feature. 

### Proposed solution

```
//--------------------------------------------------------------------------------------------------
// <auto-generated>
// "This file was generated by KontentModelGenerator v1.x.x etc"
// </auto-generated>
//--------------------------------------------------------------------------------------------------
```

Include this comment at the top of all generated files. This is something that already happens on Xperience for page types/custom tables etc.

### PR DESCRIPTION
**Pull request by jrkd**
BaseClassCodeGenerator/ClassCodeGenerator/TypeProviderCodeGenerator
- Added a comment at the top of each file wrapped in the <auto-generated> tag

ClassCodeGenerator
- Also added a separate comment for the CustomPartial version of the generated files, as while it can be edited, it is still auto generated from the tool.

### Motivation

Fixes #141 

### Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added (There arent tests for this project)
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Run the generator, and view the top of each file.

### CODE REVIEW

File: src/Kentico.Kontent.ModelGenerator.Core/BaseClassCodeGenerator.cs [hunk 1]
```diff
@@ -23,6 +23,16 @@ public class BaseClassCodeGenerator
         /// </summary>
         public string ExtenderClassName => $"{_className}Extender";
 
+        public const string FILE_COMMENT_PREFIX =
```
Comment (Sevitas) [inline src/Kentico.Kontent.ModelGenerator.Core/BaseClassCodeGenerator.cs:26]: Please use camel case

File: src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs [hunk 2]
```diff
@@ -11,6 +11,26 @@ namespace Kentico.Kontent.ModelGenerator.Core
     public class ClassCodeGenerator
     {
         public const string DEFAULT_NAMESPACE = "KenticoKontentModels";
+        public const string DEFAULT_FILE_COMMENT_PREFIX =
```
Comment (Sevitas) [inline src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs:14]: Please use camel case

File: src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs [hunk 3]
```diff
@@ -11,6 +11,26 @@ namespace Kentico.Kontent.ModelGenerator.Core
...
+// <auto-generated>
+// This code was generated by a kontent-generators-net tool 
+// (see https://github.com/Kentico/kontent-generators-net).
+// 
+// Changes to this file may cause incorrect behavior and will be lost if the code is regenerated. 
+// For further modifications of the class, create a separate file with the partial class.
+// </auto-generated>
+//--------------------------------------------------------------------------------------------------";
+
+        public const string PARTIAL_FILE_COMMENT_PREFIX =
```
Comment (Sevitas) [inline src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs:25]: Please use camel case

File: src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs [hunk 4]
```diff
@@ -13,6 +13,16 @@ namespace Kentico.Kontent.ModelGenerator.Core
     public class TypeProviderCodeGenerator
     {
         public const string CLASS_NAME = "CustomTypeProvider";
+        public const string FILE_COMMENT_PREFIX =
```
Comment (Sevitas) [inline src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs:16]: Please use camel case

File: src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs [hunk 5]
```diff
@@ -82,6 +92,9 @@ public string GetCodename(Type contentType)
 
             var cu = (CompilationUnitSyntax)tree.GetRoot();
 
+            var description = SyntaxFactory.Comment(TypeProviderCodeGenerator.FILE_COMMENT_PREFIX + Environment.NewLine + Environment.NewLine);
```
Comment (Sevitas) [inline src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs:95]: there might be useful to create a helper method for generating a comment

**Review by Sevitas — CHANGES_REQUESTED** [review]
Left few notes in the PR, however you should create a PR to the _vNext_ branch, with the latest changes.

File: src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs (hunk 2 above)
Comment (jrkd) [reply src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs]: Updated

File: src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs (hunk 3 above)
Comment (jrkd) [reply src/Kentico.Kontent.ModelGenerator.Core/ClassCodeGenerator.cs]: Updated

File: src/Kentico.Kontent.ModelGenerator.Core/BaseClassCodeGenerator.cs (hunk 1 above)
Comment (jrkd) [reply src/Kentico.Kontent.ModelGenerator.Core/BaseClassCodeGenerator.cs]: Updated

File: src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs (hunk 4 above)
Comment (jrkd) [reply src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs]: Updated

File: src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs (hunk 5 above)
Comment (jrkd) [reply src/Kentico.Kontent.ModelGenerator.Core/TypeProviderCodeGenerator.cs]: Added new file SyntaxHelpers.cs with helper method GenerateComment(string), thats used in BaseClassGenerator, ClassCodeGenerator & TypeProviderCodeGenerator

**Comment by jrkd** [comment]
> Left few notes in the PR, however you should create a PR to the _vNext_ branch, with the latest changes. There are also no tests for the functionality.

Ive updated as requested, and changed the merge target to vNext.

As for the tests, could you direct me to where tests should be placed in this project?

**Comment by Sevitas** [comment]
Hi @jrkd, 
I've looked over the changes but I think changing the target branch will not be enough. You should first resolve merge conflicts or checkout the _vNext_ branch locally and make these changes there (I think this might be easier, because there are lot of breaking changes in the project structure). 
Tests which should be edited can be found in the [https://github.com/Kentico/kontent-generators-net/tree/vNext/test/Kentico.Kontent.ModelGenerator.Tests/Generators](https://github.com/Kentico/kontent-generators-net/tree/vNext/test/Kentico.Kontent.ModelGenerator.Tests/Generators)

**Comment by Sevitas** [comment]
> > Left few notes in the PR, however you should create a PR to the _vNext_ branch, with the latest changes. There are also no tests for the functionality.
> 
> Ive updated as requested, and changed the merge target to vNext.
> 
> As for the tests, could you direct me to where tests should be placed in this project?

Hi @jrkd, 
I've looked over the changes but I think changing the target branch will not be enough. You should first resolve merge conflicts or checkout the _vNext_ branch locally and make these changes there (I think this might be easier, because there are lot of breaking changes in the project structure). 
Tests which should be edited can be found in the [https://github.com/Kentico/kontent-generators-net/tree/vNext/test/Kentico.Kontent.ModelGenerator.Tests/Generators](https://github.com/Kentico/kontent-generators-net/tree/vNext/test/Kentico.Kontent.ModelGenerator.Tests/Generators).

**Review by Sevitas — CHANGES_REQUESTED** [review]
see [comment](https://github.com/Kentico/kontent-generators-net/pull/142#issuecomment-1091333987)

**Comment by Sevitas** [comment]
Closing. The original issue #141 will be fixed by #147. Thanks for the activity @jrkd.

**Pull request closed without merging**